From 1b1430bcb773223882baa4192010c60ace5350cb Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Mon, 17 Feb 2014 09:13:42 +0000
Subject: [PATCH] builder: Fix dependencies which are not generated correctly
 by automake.

See the error messages here:
https://www.redhat.com/archives/libguestfs/2014-February/msg00148.html

This fixes commit e2cc8b6465a400024fe2f0fcce0d0ff5f7e7719c.

(cherry picked from commit 360abeebda97e9940a4045a331ce181d1a370719)
---
 builder/Makefile.am | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/builder/Makefile.am b/builder/Makefile.am
index 3087ac4..6313bad 100644
--- a/builder/Makefile.am
+++ b/builder/Makefile.am
@@ -246,9 +246,11 @@ CLEANFILES += \
 	index-scan.c \
 	stamp-virt-index-validate.pod
 
-# Fix dependencies which automake doesn't generate correctly.
 if HAVE_OCAML
-index-parser-c.o: index-parse.h index-struct.h
-index-scan.o: index-parse.h
+# Automake-generated makefile has a rule ".l.c:" but lacks a rule ".l.h:".
+# Also it doesn't generate dependencies for the C files that include
+# index-parse.h.
+index-parser-c.c index-scan.c index-validate.c: index-parse.h
+index-parse.h: index-parse.y
+	$(MAKE) index-parse.c
 endif
-index-validate.o: index-parse.h
-- 
1.8.5.3

